src/batch/src/executor/join/hash_join.rs

```diff
@@ -569,6 +578,9 @@ impl<K: HashKey> HashJoinExecutor<K> {
                     for build_row_id in
                         next_build_row_with_same_key.row_id_iter(Some(*first_matched_build_row_id))
                     {
+                        if non_equi_state.found_matched {
+                            break;
+                        }
```

Comment:
Alternative ideas:

If match is found for a probe row:
a. Either clear buffer after probe row is finished.
b. Or inside `remove_duplicate_rows_for_left_semi_anti_join` just start from `first_row_id` instead of `0`. Appended rows will get flushed anyway when buffer is spilled.
c. other ideas?

Reply:
SemiJoin should emit a row and stop probing once a matched row is found, so your modification is appropriate.